lldb/source/Commands/CommandObjectDWIMPrint.cpp

```diff
@@ -101,6 +101,10 @@ void CommandObjectDWIMPrint::DoExecute(StringRef command,
   // Add a hint if object description was requested, but no description
   // function was implemented.
   auto maybe_add_hint = [&](llvm::StringRef output) {
+    static bool note_shown = false;
+    if (note_shown)
+      return;
```

Comment:
I realize this code was already there, but this isn't great. Can we store this at the debugger level, for example? I'd expect this to show up once per debugger, and again after having called SBDebugger::Terminate/Initialize.

Reply:
If we tie it to a module, we could use the mechanism introduced in https://github.com/llvm/llvm-project/pull/112801/files to automatically show it once per session.

> I'd expect this to show up once per debugger

You mean once per type _and_ debugger, right?

Reply:
I think @augusto2112 and I discussed and decided not to do once per type. We figured that might make the hint too noisy.

Reply:
I am indifferent about per-debugger. I don't consider this note as being critical, as long as it's shown once in a while, I'm good.

Reply:
If we already landed on only once per session I'm fine with that too.
You can also manually allocate a std::once and pass it to the Debugger-wide function. Or just do the call_once here.

Reply:
Yes the idea was for it not to show up too frequently, I'd be ok if it's once per debugger too.

Reply:
I don't see how call_once would work here. This lambda might be called once, or it might be called multiple times (even indefinitely).